lib/brakeman/checks/check_unscoped_find.rb

```diff
@@ -52,12 +52,16 @@ def process_result result
       :cwe_id => [285]
   end
 
-  def optional_belongs_to? exp
+  def self_referential_association? exp, name
     return false unless exp.is_a? Array
 
     exp.each do |e|
```

Comment:
I think what we actually want is to check if there is any `belongs_to` relationship that is _not_ optional?

Is the self-referential piece actually that important?